benchexec/tools/vampire.py

```diff
@@ -0,0 +1,128 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import benchexec
+import benchexec.result as result
+
+
+class Tool(benchexec.tools.template.BaseTool2):
+    """
+    Tool info for Vampire.
+    https://github.com/vprover/vampire
+    """
+
+    def name(self):
+        return "Vampire"
+
+    def executable(self, tool_locator):
+        return tool_locator.find_executable("vampire")
+
+    def version(self, executable):
+        line = self._version_from_tool(executable, line_prefix="Vampire")
+        line = line.strip()
+        line = line.split(" ")[0]
+        return line.strip()
+
+    def environment(self, executable):
+        return {"keepEnv": {"TPTP": 1}}
+
+    def cmdline(self, executable, options, task, rlimits):
+        assert len(task.input_files) <= 1, "only one input file supported"
+
+        if "-t" not in options and "--time_limit" not in options:
+            if rlimits.walltime is None:
+                # Default timeout of Vampire is 60s,
+                # so we set value of 0 explicitly (means unlimited)
+                options += ["-t", "0"]
+            else:
+                options += ["-t", f"{rlimits.walltime}s"]
+
+        if "-m" not in options and "--memory_limit" not in options:
+            if rlimits.memory is None:
+                # No memory limit has been set
+                # TODO: should we warn in this case? Vampire is going to use the default of 3000MiB
+                pass
+            else:
+                # Vampire's option '--memory_limit/-m' takes the value in MiB
+                memory_mb = int(rlimits.memory / (1024 * 1024))
+                options += ["-m", f"{memory_mb}"]
+
+        return [executable, *options, *task.input_files]
+
+    def determine_result(self, run):
+        status = self.get_szs_status(run.output)
+        if run.exit_code:
```

Comment:
`run.exit_code´ will always be an object instance, so this condition is always true.

Reply:
I think this works as it is because `run.exit_code` is an instance of `ProcessExitCode`, which has an implementation of `__bool__`. (And my tests seemed to return the right results.)

Reply:
Right, of course you are correct. Seems you know my code better than me ;-)